### PR TITLE
refactor (core): Add a dropin "UI" for user confirmations

### PR DESCRIPTION
**Pull request by ujjwal-cyph**

### CODE REVIEW

File: apps/manager_app/device_authentication.c [hunk 1]
```diff
@@ -242,8 +243,12 @@ static device_auth_state_e sign_serial_handler(const manager_query_t *query) {
 
   switch (request_type) {
     case MANAGER_AUTH_DEVICE_REQUEST_INITIATE_TAG: {
-      // TODO: Check if it's a forced device authentication, in which case we
-      // will take users permission to perform authentication again
+      if (is_device_authenticated() &&
+          !core_user_confirmation(ui_text_start_device_verification,
+                                  manager_send_error)) {
+        // re-authentication denied by user
```
Comment (kartikCypherock) [inline apps/manager_app/device_authentication.c:249]: next_
```suggestion
        // re-authentication denied by user
        next_state = FLOW_COMPLETE;
```

**Review by kartikCypherock — APPROVED** [review]

**Comment by amanCypherock** [comment]
Please test the change in device auth before merging

File: apps/manager_app/device_authentication.c (hunk 1 above)
Comment (ujjwal-cyph) [reply apps/manager_app/device_authentication.c:247]: thanks. added

**Comment by ujjwal-cyph** [comment]
> Please test the change in device auth before merging

checked

**Pull request merged**